src/screens/InboundReceiveDetail/index.tsx

```diff
@@ -151,27 +153,47 @@ const InboundReceiveDetail: React.FC = () => {
   );
 
   const createCompleteReceivingPayload = useCallback(
-    (responseData: any) => ({
-      isShipmentFromPurchaseOrder: responseData.isShipmentFromPurchaseOrder,
-      receiptStatus: 'COMPLETED',
-      containers: responseData.containers,
-      dateShipped: responseData.dateShipped,
-      description: responseData.description,
-      destination: {
-        id: responseData['destination.id']
-      },
-      origin: {
-        id: responseData['origin.id']
-      },
-      receiptId: responseData.receiptId,
-      shipmentStatus: responseData.shipmentStatus,
-      shipment: {
-        name: responseData['shipment.name'],
-        shipmentNumber: responseData['shipment.shipmentNumber']
-      },
-      requisition: responseData.requisition,
-      recipient: responseData.recipient.id
-    }),
+    (responseData: any) => {
+      const transformedContainers = responseData.containers.map((container: any) => {
+        const {
+          'container.id': id,
```

Comment:
Long term it would be good to unflatten it in the source (API), and this part should not be needed anymore. I think I will create a ticket for doing it, because there will be more places like that. So no need to change it here.